### PR TITLE
Lps-145204

### PR DESCRIPTION
**Pull request by manuelRives**
Related Issue: https://issues.liferay.com/browse/LPS-145204

When you try to apply a field filter to an Asset Publisher, if the page is on a language different from the default locale the validation fails.

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @manuelRives, can you please check if these failures are related to your changes? Thanks

**Comment by manuelRives** [comment]
Hey @liferay-forms, I've reviewed the errors and they don't seem to be related with my changes.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js [hunk 1]
```diff
@@ -1294,7 +1294,7 @@ AUI.add(
 						var defaultLocale = instance.getDefaultLocale();
 
 						if (
-							locale === defaultLocale ||
+							!(locale in localizationMap) ||
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:1295]: Hey @manuelRives, can you please give more details about this specific change? Why did you need to do it?

Comment (manuelRives) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:1295]: Hi @natocesarrego, the old condition of locale === defaultLocale was used to allow the save when when te user has not modified the text because on that situation the localizationMap is an empty object. But the problem is that if the locale is different than the defaultLocale this condition is not fullfilled.

On the other hand with my new condition, if the locale is not stored on the locationMap, it also means that the user has not modified the text, since the localizationMap will have the avaliableLocales unless the the user has not modified the text. And on top, this condition works independently of the locale of the page, because on localized pages that Locale doesn't have to be the same one as the defaultLocale.

The lack of that second consideration was causing the problem that I identified on this LPS.

I hope this solve your question.

Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js:1295]: Gotcha, @manuelRives! It makes sense to me. :slightly_smiling_face: 

Thanks for your clarification

**Comment by natocesarrego** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @manuelRives, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @manuelRives, can you please check if these failures are related to your changes? Thanks

**Comment by manuelRives** [comment]
Hey @natocesarrego @carolmariaabb @liferay-forms 

I've reviewed the errors that are occurring on the tests and they don't seem to be related with the pull. But it seems that ci will not forward the pull automatically while they keep failing. Can you send it manually?

**Comment by natocesarrego** [comment]
@manuelRives, do you mind to rebase your pull request for one more try?

Recently, a few test fixes were merged and they could help us here. Besides, usually, we don't send pulls straightforward to Brian since we already have possible unrelated test failures affecting us later.

Thanks in advance for your understanding

**Comment by manuelRives** [comment]
Hey @natocesarrego @carolmariaabb @liferay-forms, rebased and resend.

https://github.com/liferay-forms/liferay-portal/pull/1392

Regards.
